src/wallet/wallet.cpp

```diff
@@ -4135,7 +4188,26 @@ LegacyScriptPubKeyMan* CWallet::GetOrCreateLegacyScriptPubKeyMan()
 
 void CWallet::SetupLegacyScriptPubKeyMan()
 {
-    if (!m_spk_man) m_spk_man = MakeUnique<LegacyScriptPubKeyMan>(*this);
+    if (!m_internal_spk_managers.empty() || !m_external_spk_managers.empty() || !m_spk_managers.empty()) {
+        return;
+    }
+
+    assert(m_internal_spk_managers.empty() && m_external_spk_managers.empty() && m_spk_managers.empty());
+    auto spk_manager = std::unique_ptr<ScriptPubKeyMan>(new LegacyScriptPubKeyMan(*this));
+    for (const auto& type : OUTPUT_TYPES) {
+        m_internal_spk_managers[type] = spk_manager.get();
+        m_external_spk_managers[type] = spk_manager.get();
+    }
+    m_spk_managers[spk_manager->GetID()] = std::move(spk_manager);
+
+    // These all need to exist and be the same
+    assert(m_internal_spk_managers.count(OutputType::LEGACY) > 0);
+    ScriptPubKeyMan* spk_man = m_internal_spk_managers.at(OutputType::LEGACY);
+    for (const auto& type : OUTPUT_TYPES) {
+        assert(m_internal_spk_managers.at(type) == spk_man);
+        assert(m_external_spk_managers.at(type) == spk_man);
+    }
+    assert(m_spk_managers.size() == 1);
```

Comment:
In what case would these asserts ever be hit, directly following the setup code? Is there a scenario you can think of?

Reply:
I don't think so. I've removed these asserts.

Reply:
I think a better solution would be to have this return a bool, but that's an improvement we can make in the future as it's really a belt-and-suspenders check.